### PR TITLE
Implement donkey, mule & llama inventories

### PR DESCRIPTION
**Pull request by mastercoms**
Continuation from #864

### CODE REVIEW

File: src/main/java/net/glowstone/entity/passive/GlowChestedHorse.java [hunk 1]
```diff
@@ -18,6 +33,14 @@ public boolean isCarryingChest() {
...
+        if (carryingChest != isCarryingChest()) {
+            metadata.set(MetadataIndex.CHESTED_HORSE_HAS_CHEST, carryingChest);
+            createNewInventory();
+        }
     }
+
+    /**
+     * Creates and sets a new inventory, and copies equipment over from the existing inventory.
+     */
+    protected abstract void createNewInventory();
```
Comment (aramperes) [inline src/main/java/net/glowstone/entity/passive/GlowChestedHorse.java:45]: I think this should return `<InventoryT>`, and the constructor should set the `inventory` field to the return value. The method implementations shouldn't have to update the field imo.

File: src/main/java/net/glowstone/inventory/GlowLlamaInventory.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package net.glowstone.inventory;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.LlamaInventory;
+
+public class GlowLlamaInventory extends GlowInventory implements LlamaInventory {
```
Comment (aramperes) [inline src/main/java/net/glowstone/inventory/GlowLlamaInventory.java:8]: Needs documentation

File: src/main/java/net/glowstone/entity/passive/GlowChestedHorse.java [hunk 3]
```diff
@@ -1,14 +1,29 @@
...
+import lombok.Getter;
 import net.glowstone.entity.meta.MetadataIndex;
 import org.bukkit.Location;
 import org.bukkit.entity.ChestedHorse;
 import org.bukkit.entity.EntityType;
+import org.bukkit.inventory.AbstractHorseInventory;
 
-public abstract class GlowChestedHorse extends GlowAbstractHorse implements ChestedHorse {
+/**
+ * A quadruped whose inventory may include a chest.
```
Comment (aramperes) [inline src/main/java/net/glowstone/entity/passive/GlowChestedHorse.java:11]: `quadruped` -> `horse`

"Quadruped" is too broad

**Review by aramperes — CHANGES_REQUESTED** [review]

**Comment by Pr0methean** [comment]
Superseded by #872.

**Pull request closed without merging**